src/components/chatbox/chatbox.tsx

```diff
@@ -0,0 +1,550 @@
+				}
+
+				// cleanup
+				setIsReading(false)
+			})()
+		}
+	}, [mutation.data])
+
+	return {
+		stopStream,
+		streamedText,
+		errorText,
+		conversationId,
+		messageId,
+		isLoading: mutation.isLoading || isReading,
+		sendMessage: mutation.mutate,
+	}
+}
+
+type MessageType =
+	| {
+			type: 'user'
+			image?: string
+			text: string
+	  }
+	| {
+			type: 'assistant'
+			text: string
+			messageId: string
+			conversationId: string
+	  }
+	| {
+			type: 'application'
+			text: string
+	  }
+
+const ChatBox = () => {
+	const {
+		errorText,
+		streamedText,
+		messageId,
+		conversationId,
+		isLoading,
+		stopStream,
+		sendMessage,
+	} = useAI()
+	const { user } = useAuthentication()
+
+	// Text area
+	// Note: We are not re-using the command bar's input state as we intentionally
+	// want the "search" & "AI" UX to be separate.
+	const [userInput, setUserInput] = useState('')
+
+	// List of user and assistant messages
+	const [messageList, setMessageList] = useState<MessageType[]>([])
+	const appendMessage = (message: MessageType) => {
+		setMessageList((prev) => [...prev, message])
+	}
+	const resetMessageList = () => {
+		setMessageList([])
+	}
+
+	// stream text into our message list
+	const updateAssistantMessageByIds = ({
+		conversationId,
+		messageId,
+		text,
+	}: {
+		conversationId: string
+		messageId: string
+		text: string
+	}) => {
+		setMessageList((prev) => {
+			const next = [...prev]
+			const assistantMessage = next.find(
+				(e) =>
+					e.type == 'assistant' &&
+					e.messageId === messageId &&
+					e.conversationId === conversationId
+			)
+
+			// update or create
+			if (assistantMessage) {
+				assistantMessage.text = text
+			} else {
+				next.push({
+					type: 'assistant',
+					text: text,
+					messageId: messageId,
+					conversationId: conversationId,
+				})
+			}
+			return next
+		})
+	}
+
+	// update component state when text is streamed in from the backend
+	useEffect(() => {
+		if (!streamedText || !messageId || !conversationId) {
+			return
+		}
+		updateAssistantMessageByIds({
+			conversationId,
+			messageId,
+			text: streamedText,
+		})
+	}, [streamedText, messageId, conversationId])
+
+	const handleSubmit = async (e) => {
+		const task = e.currentTarget.task?.value
+		e.preventDefault()
+
+		// Reset previous messages
+		// -- Revisit this when we're ready for multi-message conversations
+		resetMessageList()
+
+		// Clear textarea
+		setUserInput('')
+
+		sendMessage({
+			value: task,
+			// -- uncomment these parameters when we are ready for multi-message conversations
+			// conversationId,
+			// parentMessageId: messageId,
+		})
+
+		// append user message to list
+		appendMessage({ type: 'user', text: task, image: user.image })
+	}
+
+	// Throttle this value to enable uninterrupted smooth scrolling
+	const throttledText = useThrottle(streamedText, 400)
+
+	// scroll text response to bottom as text streams in from the backend
+	const textContentRef = useRef<HTMLDivElement>(null)
```

Comment:
`overflow-anchor` might be a more elegant solution here as it respects the user's ability to scroll, where as this will cause the view to scroll to the bottom even if the user scrolled up. https://css-tricks.com/books/greatest-css-tricks/pin-scrolling-to-bottom/

Reply:
Ah, I like the idea of a CSS-only-solution way better, but there's one caveat that I think is a dealbreaker / requires additional JS, making this not that much more elegant.

> _In order for scroll anchoring to work, the page must be scrolled at least once to begin with._